### ISSUE TITLE
RadioButton is not supported in dynamic forms

### ISSUE THREAD

**Issue by huineng** [primefaces/primeng#2238]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**
radio button not updated in combination with dynamic forms 

**Expected behavior**
radiobutton updated with dynamic forms 

**Minimal reproduction of the problem with instructions**

* **Angular version:** 2.4.9

* **PrimeNG version:** 2.0.2

radion button is not being checked when using dynamic form for the following reason 

https://github.com/primefaces/primeng/blob/master/components/radiobutton/radiobutton.ts

value should be checked if(this.input) 
```
writeValue(value: any) : void {
        this.checked = (value == this.value);
        
        if(this.input) {
            this.input.checked = this.checked;
        }

        this.cd.markForCheck();
    }
```

problem is that this.input is being calculated AFTER the writeValue , not before 
```
 ngAfterViewInit() {
        this.input = <HTMLInputElement> this.inputViewChild.nativeElement;
    }
```

and therefore this code "this.input.checked" **will never run** 
solution could be 
```
 if(this.inputViewChild.nativeElement) {
            this.input.checked = this.checked;
        }
```

**Comment by PeterNguyenVN** [comment]
i have the same problem , the solution is same exact 

        if (this.inputViewChild && this.inputViewChild.nativeElement) {
            this.inputViewChild.nativeElement.checked = this.checked;
        }

need to fix and notify for others when they have this problem as well.

Thanks.

**Comment by huineng** [comment]
anyone update on this ?

**Comment by Grompokstar** [comment]
I have the same problem. Need fix.

**Comment by cagataycivici** [comment]
2761 also adds that expression checked error.

**Comment by cagataycivici** [comment]
I can't replicate with;

```xml
    <form [formGroup]="userform" (ngSubmit)="onSubmit(userform.value)">
        <div *ngFor="let gender of genders">
            <p-radioButton name="gender" [value]="gender.value" [label]="gender.label" formControlName="gender"></p-radioButton>
        </div>
        
        <button pButton type="submit" label="Submit" [disabled]="!userform.valid"></button>
        
        <div style="text-align:center;margin-top:20px" *ngIf="submitted">
            Form Submitted
            <br>
            {{diagnostic}}
        </div>
    </form>
```

```typescript
export class ValidationDemo implements OnInit {
    
    msgs: Message[] = [];
    
    userform: FormGroup;
    
    submitted: boolean;
    
    genders: SelectItem[];
        
    description: string;
    
    constructor(private fb: FormBuilder) {}
    
    ngOnInit() {
        this.userform = this.fb.group({
            'gender': new FormControl('', Validators.required)
        });
        
        this.genders = [];
        this.genders.push({label:'Male', value:'Male'});
        this.genders.push({label:'Female', value:'Female'});
    }
    
    onSubmit(value: string) {
        this.submitted = true;
        this.msgs = [];
        this.msgs.push({severity:'info', summary:'Success', detail:'Form Submitted'});
    }
    
    get diagnostic() { return JSON.stringify(this.userform.value); }
    
}
```

**Comment by cagataycivici** [comment]
The current code has;

    ngAfterViewInit() {
        this.input = <HTMLInputElement> this.inputViewChild.nativeElement;
    }

So probably that fixes this issue, moving attention to #2761
